packages/runtime/container-runtime/src/opLifecycle/opSplitter.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { ContainerMessageType } from "../containerRuntime";
+import { IChunkedOp } from "./definitions";
+
+/**
+ * Responsible for creating and reconstructing chunked messages.
+ */
+export class OpSplitter {
+    // Local copy of incomplete received chunks.
+    private readonly chunkMap: Map<string, string[]>;
+
+    constructor(chunks: [string, string[]][]) {
+        this.chunkMap = new Map<string, string[]>(chunks);
+    }
+
+    public get hasChunks(): boolean {
```

Comment:
Not sure if this API is needed - it feels like just calling chunks() and checking if it's empty by caller is sufficient.

Reply:
That said, see my comment below on snapshotting